src/com/swdc/codetime/models/KeystrokeCountUtil.java

```diff
@@ -0,0 +1,165 @@
+package com.swdc.codetime.models;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.swdc.codetime.managers.EclipseProjectUtil;
+import com.swdc.codetime.managers.FileAggregateDataManager;
+import com.swdc.codetime.managers.SessionDataManager;
+import com.swdc.codetime.managers.TimeDataManager;
+import com.swdc.codetime.managers.WallClockManager;
+import com.swdc.codetime.util.SoftwareCoUtils;
+
+import swdc.java.ops.manager.EventTrackerManager;
+import swdc.java.ops.manager.FileUtilManager;
+import swdc.java.ops.manager.GitUtilManager;
+import swdc.java.ops.manager.UtilManager;
+import swdc.java.ops.model.CodeTime;
+import swdc.java.ops.model.CodeTime.FileInfo;
+import swdc.java.ops.model.ElapsedTime;
+import swdc.java.ops.model.FileChangeInfo;
+import swdc.java.ops.model.Project;
+import swdc.java.ops.model.ResourceInfo;
+import swdc.java.ops.model.TimeData;
+
+public class KeystrokeCountUtil {
+
+	public static long cumulative_editor_seconds = 0;
+	public static long elapsed_seconds = 0;
+	public static String project_null_error = "";
+
+	public static void processKeystrokes(CodeTime keystrokeCountInfo) {
+		if (keystrokeCountInfo.hasData()) {
+
+			// make sure we have a valid project
+			Project activeProject = EclipseProjectUtil.getInstance().getFirstActiveProject();
+			if (keystrokeCountInfo.getProject() == null
+					|| StringUtils.isBlank(keystrokeCountInfo.getProject().getDirectory())) {
+				keystrokeCountInfo.setProject(activeProject);
+			}
+
+			// get the resource identifier info
+			if (!keystrokeCountInfo.getProject().getDirectory().equals(UtilManager.untitled_file_name)) {
+				// get the resource information
+				ResourceInfo resourceInfo = GitUtilManager.getResourceInfo(keystrokeCountInfo.getProject().getDirectory());
+				if (resourceInfo != null) {
+					keystrokeCountInfo.getProject().setResource(resourceInfo);
+					keystrokeCountInfo.getProject().setIdentifier(resourceInfo.getIdentifier());
+				}
+			}
+
+			ElapsedTime eTime = SessionDataManager.getTimeBetweenLastPayload();
+
+			// end the file end times.
+			preProcessKeystrokeData(keystrokeCountInfo, eTime.sessionSeconds, eTime.elapsedSeconds);
+
+			// update the file aggregate info.
+			updateAggregates(keystrokeCountInfo, eTime.sessionSeconds);
+
+			// send the event to the event tracker
+			EventTrackerManager.getInstance().trackCodeTimeEvent(keystrokeCountInfo);
+
+			// update the payload end timestmap
+			UtilManager.TimesData timesData = UtilManager.getTimesData();
+			FileUtilManager.setNumericItem("latestPayloadTimestampEndUtc", timesData.now);
+		}
+
+		keystrokeCountInfo.resetData();
+
+		WallClockManager.getInstance().dispatchStatusViewUpdate();
+	}
+
+	private static void validateAndUpdateCumulativeData(CodeTime keystrokeCountInfo, long sessionSeconds) {
```

Comment:
is all of this logic still needed? should we use the current day cached values?

Reply:
we'll have to subscribe to that event. I can add that next